packages/api/src/utils/login.ts

```diff
@@ -231,9 +231,9 @@ export abstract class BasicLogin {
     return ok(true);
   }
 
-  async removeStatusChangeMap(name: string): Promise<Result<boolean, FxError>> {
+  removeStatusChangeMap(name: string): Promise<Result<boolean, FxError>> {
     this.statusChangeMap.delete(name);
-    return ok(true);
+    return Promise.resolve(ok(true));
```

Comment:
What's the effort if we also update the return type from `Promise<Result<boolean, FxError>>` to `Result<boolean, FxError>`? Seems there's no need to return a promise here.

Reply:
This method is in the M365TokenProvider interface.
There are concrete implementations in vscode/cli.
We need to change them to keep the method interface consistant.

Reply:
![Uploading image.png…]()